src/inference/src/ie_core.cpp

```diff
@@ -494,13 +494,45 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             res = LoadNetworkFromCache(cacheManager, hash, plugin, parsed._config, context, loadedFromCache);
             if (!loadedFromCache) {
                 res = compile_model_impl(network, plugin, parsed._config, context, hash);
+            } else {
+                updateCNNInfo(res, network);
             }
         } else {
             res = compile_model_impl(network, plugin, parsed._config, context, {});
         }
         return res;
     }
 
+    static void updateCNNInfo(ie::SoExecutableNetworkInternal& exe_net, const ie::CNNNetwork& network) {
+        // Temporary workaround until all plugins support caching of original model inputs
+        OPENVINO_SUPPRESS_DEPRECATED_START
+        exe_net->setNetworkInputs(network.getInputsInfo());
+        exe_net->setNetworkOutputs(network.getOutputsInfo());
```

Comment:
I think we need to perform a deep copy, because if I change `network` in user's code, `exe_net` will be changed as well 

see https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp#L52-L76